lib/rucio/core/did.py

```diff
@@ -1829,3 +1830,157 @@ def list_archive_content(scope, name, session=None):
+
+    :param dids: A list of dids.
+    :param account: The account owner.
+    :param session: The database session in use.
+    """
+    try:
+        for did in dids:
+            session.query(models.DidsFollowed).\
+                filter_by(scope=did['scope'], name=did['name'], account=account).\
+                delete(synchronize_session=False)
+
+    except NoResultFound:
+        raise exception.DataIdentifierNotFound("Data identifier '%s:%s' not found" % (did['scope'], did['name']))
+
+
+@transactional_session
+def trigger_event(scope, name, event_type, payload, session=None):
+    """
+    Records changes occuring in the did to the FollowEvents table
+
+    :param scope: The scope name.
+    :param name: The data identifier name.
+    :param event_type: The type of event affecting the did.
+    :param payload: Any message to be stored along with the event.
+    :param session: The database session in use.
+    """
+    try:
+        dids = session.query(models.DidsFollowed).filter_by(scope=scope, name=name).all()
+
+        for did in dids:
+            # Create a new event using teh specified parameters.
+            new_event = models.FollowEvents(scope=scope, name=name, account=did.account,
+                                            did_type=did.did_type, event_type=event_type, payload=payload)
+            new_event.save(session=session, flush=False)
+
+        session.flush()
+    except IntegrityError as error:
+        raise exception.RucioException(error.args)
+
+
+@read_session
+def create_reports(session=None):
+    """
+    Create a summary report of the events affecting a dataset, for its followers.
+
+    :param session: The database session in use.
+    """
+    # Get all the distinct accounts from the table.
+    accounts = session.query(models.FollowEvents.account).distinct()
+    for account in accounts:
+        try:
+            events = session.query(models.FollowEvents).filter_by(account=account).all()
+            # If events exist for an account then create a report.
+            if events:
+                body = '''
+                       Hello,
+                       This is an auto-generated report of the events that have affected the datasets you follow.
+
+                       '''
+                # Add each event to the message body.
+                for i, event in enumerate(events):
+                    body += "{}. Dataset: {} Event: {}\n".format(i, event.name, event.event_type)
+                    if event.payload:
+                        body += "Message: {}\n".format(event.payload)
+                    body += "\n"
+
+                body += "Thank You."
+
+                # Get the email associated with the account.
+                email = session.query(models.Account.email).filter_by(account=account)
+                add_message('email', {'to': email,
+                                      'subject': 'Report of affected dataset(s)',
+                                      'body': body})
+
```

Comment:
One should clean the events when the message was successfully sent, otherwise at the next iteration the same list of events will be sent. Another possibility is to keep all the events, but only send the recent ones, based on created_at.